### PR TITLE
rpcdaemon: page_token management in Data API Range ops

### PR DESCRIPTION
**Pull request by Sixtysixter**
Fix `page_token` management in Range queries: add `page_token` parameter to `PaginatedSequence::Paginator` and `PaginatedSequence::PageResult`.

### CODE REVIEW

File: silkworm/db/kv/api/endpoint/paginated_sequence.hpp [hunk 1]
```diff
@@ -119,9 +119,9 @@ class PaginatedSequencePair {
     struct PageResult {
         KPage keys;
         VPage values;
-        bool has_more{false};
+        std::string next_page_token;
     };
-    using Paginator = std::function<Task<PageResult>()>;
+    using Paginator = std::function<Task<PageResult>(const std::string& page_token)>;
```
Comment (canepat) [inline silkworm/db/kv/api/endpoint/paginated_sequence.hpp:124]:
```suggestion
    using Paginator = std::function<Task<PageResult>(std::string)>;
```

File: silkworm/db/kv/api/endpoint/paginated_sequence_test.cpp [hunk 2]
```diff
@@ -181,7 +190,14 @@ TEST_CASE_METHOD(PaginatedSequenceTest, "set_intersection", "[db][kv][api][pagin
...
             TestPaginatorUint64 paginator1{v1}, paginator2{v2};
-            PaginatedUint64 paginated1{paginator1}, paginated2{paginator2};
+            auto lambda1 = [&](const std::string&) mutable -> Task<PageResultUint64> {
+                co_return co_await paginator1();
+            };
+            auto lambda2 = [&](const std::string&) mutable -> Task<PageResultUint64> {
+                co_return co_await paginator2();
+            };
+
+            PaginatedUint64 paginated1{lambda1}, paginated2{lambda2};
```
Comment (canepat) [inline silkworm/db/kv/api/endpoint/paginated_sequence_test.cpp:200]: This change seems unnecessary

File: silkworm/db/kv/api/endpoint/paginated_sequence_test.cpp [hunk 3]
```diff
@@ -205,7 +221,14 @@ TEST_CASE_METHOD(PaginatedSequenceTest, "set_union", "[db][kv][api][paginated_se
...
```
Comment (canepat) [inline silkworm/db/kv/api/endpoint/paginated_sequence_test.cpp:231]: This change seems unnecessary

File: silkworm/db/kv/api/endpoint/paginated_sequence_test.cpp [hunk 4]
```diff
@@ -87,20 +92,24 @@ TEST_CASE_METHOD(PaginatedSequenceTest, "paginated_uint64_sequence: non-empty se
     for (const auto& [page_list, expected_sequence] : fixtures) {
         SECTION("test vector: " + std::to_string(++i)) {
             TestPaginatorUint64 paginator{page_list};
-            PaginatedUint64 paginated{paginator};
+            auto lambda = [&](const std::string&) mutable -> Task<PageResultUint64> {
+                co_return co_await paginator();
+            };
+
+            PaginatedUint64 paginated{lambda};
```
Comment (canepat) [inline silkworm/db/kv/api/endpoint/paginated_sequence_test.cpp:99]: This change seems unnecessary

File: silkworm/db/kv/api/endpoint/paginated_sequence_test.cpp [hunk 5]
```diff
@@ -66,7 +67,11 @@ struct TestPaginatorUint64 {
 TEST_CASE_METHOD(PaginatedSequenceTest, "paginated_uint64_sequence: empty sequence", "[db][kv][api][paginated_sequence]") {
     PageUint64List empty;
     TestPaginatorUint64 paginator{empty};
-    PaginatedUint64 paginated{paginator};
+    auto lambda = [&](const std::string&) mutable -> Task<PageResultUint64> {
+        co_return co_await paginator();
+    };
+
+    PaginatedUint64 paginated{lambda};
```
Comment (canepat) [inline silkworm/db/kv/api/endpoint/paginated_sequence_test.cpp:74]: This change seems unnecessary

**Review by canepat — CHANGES_REQUESTED** [review]

File: silkworm/db/kv/api/endpoint/paginated_sequence.hpp [hunk 6]
```diff
@@ -48,9 +48,9 @@ class PaginatedSequence {
     using Page = std::vector<T>;
     struct PageResult {
         Page values;
-        bool has_more{false};
+        std::string next_page_token;
     };
-    using Paginator = std::function<Task<PageResult>()>;
+    using Paginator = std::function<Task<PageResult>(const std::string& page_token)>;
```
[the remark this answers is not in the document]
Comment (Sixtysixter) [reply silkworm/db/kv/api/endpoint/paginated_sequence.hpp:53]: done

File: silkworm/db/kv/api/endpoint/paginated_sequence.hpp (hunk 1 above)
Comment (Sixtysixter) [reply silkworm/db/kv/api/endpoint/paginated_sequence.hpp:124]: Done

File: silkworm/db/kv/api/endpoint/paginated_sequence_test.cpp (hunk 3 above)
Comment (Sixtysixter) [reply silkworm/db/kv/api/endpoint/paginated_sequence_test.cpp:231]: as previous comment

File: silkworm/db/kv/api/endpoint/paginated_sequence_test.cpp (hunk 5 above)
Comment (Sixtysixter) [reply silkworm/db/kv/api/endpoint/paginated_sequence_test.cpp:74]: as previous comment

File: silkworm/db/kv/api/endpoint/paginated_sequence_test.cpp (hunk 4 above)
Comment (Sixtysixter) [reply silkworm/db/kv/api/endpoint/paginated_sequence_test.cpp:99]: done

**Review by canepat — APPROVED** [review]

**Pull request merged**